source/digits_hits/src/GateRejectedVolume.cc

```diff
@@ -0,0 +1,68 @@
+/*----------------------
+   Copyright (C): OpenGATE Collaboration
+
+This software is distributed under the terms
+of the GNU Lesser General  Public Licence (LGPL)
+See LICENSE.md for further details
+----------------------*/
+
+
+#include "GateRejectedVolume.hh"
+
+#include "G4UnitsTable.hh"
+
+#include "GateRejectedVolumeMessenger.hh"
+#include "GateTools.hh"
+
+GateRejectedVolume::GateRejectedVolume(GatePulseProcessorChain* itsChain,
+			       const G4String& itsName,
+      	      	      	      	 G4String rejectedVolume)
+  : GateVPulseProcessor(itsChain,itsName),
+    m_rejectedVolume(rejectedVolume)
+{
+  m_messenger = new GateRejectedVolumeMessenger(this);
+}
+
+
+
+
+GateRejectedVolume::~GateRejectedVolume()
+{
+  delete m_messenger;
+}
+
+
+
+void GateRejectedVolume::ProcessOnePulse(const GatePulse* inputPulse,GatePulseList& outputPulseList)
+{
+  if (!inputPulse) {
+    if (nVerboseLevel>1)
+      	G4cout << "[GateRejectedVolume::ProcessOnePulse]: input pulse was null -> nothing to do\n\n";
+    return;
+  }
+  if (((inputPulse->GetVolumeID()).GetBottomCreator())->GetObjectName() == "") {
+    if (nVerboseLevel>1)
+      	G4cout << "[GateRejectedVolume::ProcessOneHit]: volume is null for " << inputPulse << " -> pulse ignored\n\n";
```

Comment:
Isn't there already some function defined in Gate that checks of ``nVerboseLevel`` before printing? If not maybe create one in this class, as you checking ``nVerboseLevel`` at least 4 times.

Reply:
I am not sure if there is any function in Gate which check the verbosity level, however I did a quick look for a few other classes and this kind of check is common.